lib/engine/game/g_18_norway/steps/buy_train.rb

```diff
@@ -41,11 +41,25 @@ def must_buy_train?(entity)
           end
 
           def buyable_trains(entity)
-            super.reject do |train|
-              next false unless @game.ship?(train)
+            # We need to add the ships in the case that the company have a train but have less money then the next available train
+            ships = if must_buy_train?(entity) || entity.cash >= @depot.min_depot_price
+                      []
+                    else
+                      @depot.depot_trains.reject do |train|
+                        true unless @game.ship?(train)
+                        true if train.price > entity.cash
```

Comment:
Does this work? I would have thought that the first line in the block would do nothing, shouldn't it be `next true unless @game.ship?(train)`? Or you could rewrite the block as:
```suggestion
                        @game.ship?(train) || train.price > entity.cash
```

Reply:
It looked like it worked but I have compacted the implementation a bit